Sources/_NIOFileSystem/Internal/System Calls/Syscall.swift

```diff
@@ -120,13 +123,22 @@ public enum Syscall: Sendable {
         nothingOrErrno(retryOnInterrupt: false) {
             old.withPlatformString { oldPath in
                 new.withPlatformString { newPath in
+                #if os(FreeBSD)
+                    system_renameat(
+                        oldFD.rawValue,
+                        oldPath,
+                        newFD.rawValue,
+                        newPath
+                    )
+                #else
```

Comment:
It's not apparent to me that ignoring the flags is appropriate here. Should this be instead unavailable on FreeBSD and replaced at the various call-sites with the more appropriate call based on the values of the flags?